receiver/dcgmreceiver/scraper.go

```diff
@@ -216,14 +216,19 @@ func (s *dcgmScraper) pollClient(ctx context.Context, client *dcgmClient, metric
 			)/2,
 		)
 		s.settings.Logger.Sugar().Debugf("Waiting %s for the next collection", waitTime)
-		deviceMetrics := client.getDeviceMetrics()
-		select {
-		case <-ctx.Done():
-			return
-		case <-collectTriggerCh:
-			// Loop and trigger a collect() again.
-		case metricsCh <- deviceMetrics:
-		case <-time.After(waitTime):
+		after := time.After(waitTime)
+		for after != nil {
+			deviceMetrics := client.getDeviceMetrics()
```

Comment:
I may have asked this before, but are we going to call `client.getDeviceMetrics()` unconditionally, even if nobody is reading from `metricsCh`? If so, can we only call it on-demand?

Reply:
Yes, we discussed this on one of the video chats. We said that it would be premature optimization to avoid it.

There's no way in Go to "check if a channel has a waiting receiver" without trying to send on it, and to send on it you need to already have the value you're sending. To avoid that construction we'd need to introduce a higher level of synchronization (for instance, use a channel of channels).